### PR TITLE
2480 - Delete old migration system and migrations

### PR DESCRIPTION
**Pull request by sorja**
Note: Test running disable for this PR in test.js.yml
Next PR will focus on this solely

- delete old / deprecated db instance: `src/server/db/db_deprecated.ts`
- delete old migrations and references: `src/server/db/migration/migrations/`
- delete old migration script: `src/server/db/migration/`

Note this migration step is ran before the previous pull requests migration as so:
image

We clear the migration table in this step and next step will populate it with given steps

DB Output:

image
image

### CODE REVIEW

File: src/test/migrations/steps/20240909101346-step-clear-migrations-table.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { BaseProtocol } from 'server/db'
```
Comment (sorja) [inline src/test/migrations/steps/20240909101346-step-clear-migrations-table.ts:1]: this is relevant file

File: package.json [hunk 2]
```diff
@@ -9,17 +9,15 @@
     "build": "run-s build:prod:client build:prod:server build:prod:server:resources",
```
Comment (sorja) [inline package.json:9]: this is relevant file

File: README.md [hunk 3]
```diff
@@ -29,7 +29,7 @@ To constantly build it when something changes, run:
 
 ## Database
```
Comment (sorja) [inline README.md:30]: relevant file

File: .github/workflows/test.js.yml [hunk 4]
```diff
@@ -55,13 +55,6 @@ jobs:
             NODE_AUTH_TOKEN: ${{ secrets.ACCESS_TOKEN }}
       - name: Install Playwright Browsers
```
Comment (sorja) [inline .github/workflows/test.js.yml:56]: relevant file

File: .github/workflows/test.js.yml [hunk 5]
```diff
@@ -55,28 +55,21 @@ jobs:
...
           PGPASSWORD: frap
-      - run: yarn test
-        env:
-          CI: true
-          PGHOST: localhost
-          PGPORT: 5442
-          PGDATABASE: frap-dev
-          PGUSER: frap
-          PGPASSWORD: frap
+#      - run: yarn test
```
Comment (sorja) [inline .github/workflows/test.js.yml:65]: next pr will re-enable this

Comment (minotogna) [reply .github/workflows/test.js.yml:65]: Why is it disabled now ?

Comment (sorja) [reply .github/workflows/test.js.yml:65]: Some changes required to get it working again, prefer to have these changes in another PR as they are not related to this PR

Comment (minotogna) [reply .github/workflows/test.js.yml:65]: Ok 👌

File: package.json [hunk 6]
```diff
@@ -9,24 +9,22 @@
...
     "build:prod:client": "cross-env NODE_ENV=production webpack --config webpack.config.babel.js",
     "build:prod:server": "yarn tsc --project tsconfig.server.json && tsc-alias -p tsconfig.server.json",
-    "build:prod:server:resources": "shx cp -Rf src/server/db/migration/* dist/server/db/migration/. && shx cp -Rf web-resources dist/ && shx cp -Rf src/server/static dist/server/",
+    "build:prod:server:resources": "shx cp -Rf web-resources dist/ && shx cp -Rf src/server/static dist/server/",
     "build:prod:report": "cross-env NODE_ENV=production BUILD_REPORT=true webpack --display-error-details -p --config webpack.config.babel.js",
     "clean": "shx rm -rf dist/",
     "heroku-prebuild": "echo \"//npm.pkg.github.com/:_authToken=${NPM_GITHUB_TOKEN}\" >> .npmrc",
     "heroku-cleanup": "cd node_modules/bcrypt && node-pre-gyp install --fallback-to-build",
-    "migrations:create": "./src/server/db/migration/create-migration.sh",
-    "migrations:run": "./src/server/db/migration/run-migrations.sh",
```
Comment (minotogna) [inline package.json:18]: Do we need to remove some old dependency?

Comment (sorja) [reply package.json:18]: Yes, removed `db-migrate`, `db-migrate-pg`, good catch

**Review by minotogna — APPROVED** [review]

**Pull request merged**